src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> node(int index) {
+        Node<T> find;
+        if (index < (size / 2)) {
+            find = head;
+            for (int i = 0; i < index; i++) {
+                find = find.next;
+            }
+        } else {
+            find = tail;
+            for (int i = size - 1; i > index; i--) {
+                find = find.prev;
+            }
+        }
+        return find;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> t = tail;
+        Node<T> newNode = new Node<>(t, value, null);
+        tail = newNode;
+        if (t == null) {
+            head = newNode;
+        } else {
+            t.next = newNode;
+        }
+        size++;
+        return true;
```

Comment:
how about this))
```suggestion
        Node<T> newNode = new Node<>(tail, value, null);
        if (tail == null) {
            head = newNode;
        } else {
            tail.next = newNode;
        }
        tail = newNode;
        size++;
        return true;
```

Reply:
:DDD ok you broke my logic. I don’t know why I didn’t do it, thks))